src/cosmic_ray/interceptors/spor.py

```diff
@@ -29,7 +28,7 @@ def file_contents(file_path):
             with file_path.open(mode="rt") as handle:
                 return handle.readlines()
 
-        for item in self.work_db.work_items:
+        for item in self.work_db.pending_work_items:
```

Comment:
This is incorrect. Interceptors are very simple: after init, they get to manipulate the workdb. They don't care about pending work items, only all work items. 

And in any event, you shouldn't need to modify the spor interceptor to implement your own.

Reply:
pending_work_item filters all work items with results. Then I filter work_item potentially already skipped by another interceptor.

Reply:
pending_work_item will filter already skipped item. If another interceptor has already skipped a work_item, it is useless to analyze it again.